refactored_homeWork

Burito2021 opened the pull request.

Burito2021 committed:
-create Service, Storage, Application
-add Faker library
-correct gradle dependencies, gradle java version
-add unit tests
-move methods from Storage to Service class
-add gradle.properties file

MaximDzhezhelo commented on build.gradle:
```diff
@@ -16,14 +18,16 @@ repositories {
 dependencies {
     implementation("org.junit.jupiter:junit-jupiter-api:$junit")
     implementation("com.github.javafaker:javafaker:$faker")
+    implementation("org.junit.jupiter:junit-jupiter-params:$junit")
```
Why do you add the junit to this dependency type?

MaximDzhezhelo commented on src/main/java/net/education/kyivstar/Application.java:
```diff
@@ -1,32 +1,35 @@
 
         System.out.println("Get by surname method: " + service.getBySurname("Dzhezhelo"));
-
+        System.out.println("List "+ service.printStorage());
```
Use logger for these purposes

MaximDzhezhelo commented on src/main/java/net/education/kyivstar/Repository.java:
```diff
@@ -0,0 +1,41 @@
+
+class Repository {
+    private Logger logger = LoggerFactory.getLogger(Service.class);
+    Storage data = new Storage();
```
Initialize this object throughout the constructor

MaximDzhezhelo commented on src/main/java/net/education/kyivstar/Service.java:
```diff
@@ -5,22 +5,24 @@
     private final Faker faker;
-    private final Storage store;
     private final Random random;
+    Repository repository = new Repository();
```
Initialize this object throughout the constructor

MaximDzhezhelo commented on src/main/java/net/education/kyivstar/Storage.java:
```diff
@@ -7,9 +7,9 @@
 
-    protected Storage() {
+    public Storage() {
         this.storage = new ArrayList<>();
```
Initialize this object throughout the constructor

MaximDzhezhelo commented on src/test/java/net/education/kyivstar/RepositoryTest.java:
```diff
@@ -0,0 +1,74 @@
+        assertEquals("NameReviser", actual.get(0).getName());
+        assertEquals("SurnameReviser", actual.get(0).getSurname());
+        assertEquals(30, actual.get(0).getAge());
+    }
```
You should clean up your repository at the end of each test

Burito2021 committed:
-initialize Objects in Services, Repository through Constructor
-add logging according to commmets
-delete unnecessary junit dependency
-add remove in each test

Burito2021 committed:
-divide Repository class into Teacher, Student, Reviser repositories
-create tests for UserServiceTest
-rename Storage methods
-add validations to methods
-change methods implementation from manual taverse to streams
-move some methods from UserService to Repository mthod
-all Assertions changed to 'org.junit.jupiter.api.Assertions'
